include/hw_unique_key.rst

```diff
@@ -0,0 +1,81 @@
+.. _lib_hw_unique_key:
+
+Hardware unique key
+###################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The hardware unique key library manages symmetric root keys on devices with the Arm CryptoCell security subsystem.
+
+Functionality
+*************
+
+Hardware unique keys (HUKs) are device-specific keys that can be used with functions for key derivation to generate other keys.
+The generated keys can then be used by applications for symmetric encryption or authentication.
+By using HUKs, you can let your application use multiple keys without having to store them on the device, as they can be recreated from a HUK when needed.
+
+.. note::
+   You should use hardware unique keys only for key derivation, but never directly for symmetric encryption or authentication.
+
+The library supports up to 3 different HUKs:
+
+* Device root key (KDR).
+  It can be used for deriving general-purpose keys.
+  It must be written to a specific KDR register in the Arm CryptoCell CC310.
+* Master key encryption key (MKEK).
+  It is used for deriving Key Encryption Keys (KEKs), which are used to encrypt other keys when these are stored.
+* Master key for encrypting external storage (MEXT).
+  It is used to derive keys for encrypting data in external non-secure storage.
+
+See the following table for an overview of the key types supported by each device:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Device
+      - CryptoCell version
+      - Key Management Unit
+      - Supported HUK types
+    * - nRF9160
+      - CC310
+      - Yes
+      - KDR, MKEK, MEXT
+    * - nRF5340
+      - CC312
+      - Yes
+      - MKEK, MEXT
+    * - nRF52840
+      - CC310
+      - No
+      - KDR only
+
+In devices with a Key Management Unit (KMU), like nRF9160 or nRF5340, the keys reside in reserved slots in the KMU itself.
+The KMU can make the keys non-readable and non-writable from the application, but accessible by the Arm CryptoCell.
+
+In devices without a KMU, like nRF52840, the bootloader writes the key to the Arm CryptoCell and locks the flash memory page where the key is stored.
+In this case, only one key is supported.
+
```

Comment:
We should probably have details here about how to use the keys via cc3xx_platform, or at least link to some documentation.

Reply:
@FrancescoSer @frkv added link to cc3xx_platform readme.